modules/machineset-creating-azure-ultra-disk.adoc

```diff
@@ -0,0 +1,285 @@
+
+.Prerequisites
+
+* Have an existing Microsoft Azure cluster.
+
+.Procedure
+
+ifdef::mapi[]
+. Create a new custom secret using the worker-data-secret by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get secret worker-user-data \
+--template='{{index .data.userData | base64decode}}' | jq > userData.txt
+----
++
+where #need to understand what these values are and where they come from#
+
+. Edit the `userData.txt` file by running the following command: #need this command#
++
+[source,terminal]
+----
+$ oc edit #????#
+----
++
+where #values and what they are#
+
+. Locate the final `}` in the `userData.txt` file. On the immediately preceding line, add a `,`. Create a new line after the `,` and add the following configuration details:
++
+[source,json]
+----
+"storage": {
+  "disks": [
+    {
+      "device": "/dev/disk/azure/scsi1/lun0",
+      "partitions": [
+        {
+          "label": "lun0p1",
+          "sizeMiB": 1024,
+          "startMiB": 0
+        }
+      ]
+    }
+  ],
+  "filesystems": [
+    {
+      "device": "/dev/disk/by-partlabel/lun0p1",
+      "format": "xfs",
+      "path": "/var/lib/lun0p1"
+    }
+  ]
+},
+"systemd": {
+  "units": [
+    {
+      "contents": "[Unit]\nBefore=local-fs.target\n[Mount]\nWhere=/var/lib/lun0p1\nWhat=/dev/disk/by-partlabel/lun0p1\nOptions=defaults,pquota\n[Install]\nWantedBy=local-fs.target\n",
+      "enabled": true,
+      "name": "var-lib-lun0p1.mount"
+    }
+  ]
+}
+----
++
+where #need to understand what these values are and where they come from#
+
+. Extract the disabling template value by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get secret worker-user-data --template='{{index .data.disableTemplating | base64decode}}' | jq > disableTemplating.txt
+----
++
+where #need to understand what these values are and where they come from#
+
+. Merge the two files to create a datasecret file to be used by running the following command:
++
+[source,terminal]
+----
+oc -n openshift-machine-api create secret generic worker-user-data-x5 \
+--from-file=userData=userData.txt \
+--from-file=disableTemplating=disableTemplating.txt
+----
++
+where #need to understand what these values are and where they come from#
+endif::mapi[]
+
+. Edit or copy an existing Azure `MachineSet` custom resource (CR) by running the following command:
++
+[source,terminal]
+----
+$ oc edit machineset <machine-set-name>
+----
++
+where `<machine-set-name>` is the machine set that you want to provision machines on ultra disks.
+
+. Add the following lines in the positions indicated:
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+  ...
+spec:
+  ...
+  template:
+    ...
+    spec:
+      metadata:
+        ...
+        labels:
+          ...
+ifdef::mapi[          disktype: ultrassd <1>]
+ifdef::pvc[          disk: ultrassd <1>]
```

Comment:
I think we could consolidate these labels.
I created them different to avoid collision when testing, but for separate examples they could be the same. WDYT?

Reply:
I wondered about the difference there. I would be delighted to make them the same, just tell me which one we want :)

Reply:
I'd go with `disk: ultrassd` :) as it's shorter.